http/netty/components/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/sender/TargetHandler.java

```diff
@@ -123,7 +132,7 @@ protected void readTimedOut(ChannelHandlerContext ctx) {
         ctx.channel().close();
 
         if (targetChannel.isRequestWritten()) {
-            String payload = "<errorMessage>" + "ReadTimeoutException occurred for endpoint" + targetChannel.
+            String payload = "<errorMessage>" + "ReadTimeoutException occurred for endpoint " + targetChannel.
```

Comment:
Even though we set a XML payload, we can configure the MIME type to be application/json or any other. How it works is this payload gets injected to camel exchange as an exception. When there is an exception in camel exchange, that exception can be handled using camel spring configuration as follows. 

<onException>
   <exception>org.wso2.carbon.messaging.exceptions.EndPointTimeOut</exception>
   <handled>
      <constant>true</constant>
   </handled>
   <process ref="failureResponse" />
</onException>

<bean id="failureResponse" class="org.wso2.carbon.gateway.internal.mediation.camel.CustomErrorHandlingProcessor">
   <property name="errorMessage" value="Back-end server did not response withing the configured timeout" />
   <property name="statusCode" value="500" />
   <property name="contentType" value="application/xml" />
</bean>